### ISSUE TITLE
nextfloat()/prevfloat() does not satisfy documentation when set_zero_subnormals(true)

### ISSUE THREAD

**Issue by amckinlay** [JuliaLang/julia#18652]
```
julia> set_zero_subnormals(true)
true

julia> 0. < nextfloat(0.)
false

julia> 0. == nextfloat(0.)
true

julia> 0. === nextfloat(0.)
false
```

```
help?> nextfloat(0.)
  nextfloat(x::AbstractFloat)

  Returns the smallest floating point number y of the same type as x such x < y. If no such y exists (e.g. if x is Inf or NaN), then returns x.
```

```
julia> versioninfo()
Julia Version 0.5.0
Commit 3c9d753 (2016-09-19 18:14 UTC)
Platform Info:
  System: Darwin (x86_64-apple-darwin13.4.0)
  CPU: Intel(R) Core(TM) i5-5257U CPU @ 2.70GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.1 (ORCJIT, broadwell)
```

**Comment by yuyichao** [comment]
They are doing the right thing since the next floating point is a subnormal. It's another story that the hardware swallows the return value since that's what you are asking for.

**Comment by yuyichao** [comment]
In another word, the document says that the function `Returns the smallest floating point number y of the same type as x such x < y` not `Returns the smallest floating point number y of the same type **supported by the current hardware setting** as x such x < y`.

**Comment by simonbyrne** [comment]
I think this sentence covers it:
> Otherwise, floating-point operations are permitted (but not required) to convert subnormal inputs or outputs to zero
which seems to give fairly wide latitude to operations involving subnormals. 

I don't think we can change the behaviour here without a large performance penalty (and performance is the main point of FZ/DAZ mode). Though if you think the docs can be improved, please do open a PR (and ping me on it). If you do, would you mind also changing the argument name while you're at it to avoid this rather confusing part:
>  If `yes` is `false`, ...

**Comment by thchr** [comment]
It would have been nice to mention this in the docs of `prevfloat` and `nextfloat` though.

As is, the behavior is likely to be surprising for many people. The docs currently promise that `prevfloat(x)` will return `y` such that `y < x`, but that is not so for `prevfloat(0.0)`:

```jl
julia> x = 0.0;
julia> y = prevfloat(x)
-0.0

julia> y < x
false
```
There is a related surprise in that `prevfloat(x, n) has a fixed point for `x = -0.0` (i.e., always returns `-0.0`, regardless of `n`). Similarly for `nextfloat(0.0, n)`
